src/repositoryModals.jsx

```diff
@@ -0,0 +1,451 @@
+    );
+};
+
+EditRepositoryModal.propTypes = {
+    remote: PropTypes.string.isRequired,
+    availableRemotes: PropTypes.array.isRequired,
+};
+
+export const RebaseRepositoryModal = ({ origin, availableRemotes, currentOriginBranches, currentBranchLoadError, onChangeBranch, onChangeRemoteOrigin }) => {
+    const Dialogs = useDialogs();
+    const [isSelectRemoteExpanded, setSelectRemoteExpanded] = useState(false);
+    const [selectedRemote, setSelectedRemote] = useState(origin.remote);
+    const [isSelectBranchExpanded, setSelectBranchExpanded] = useState(false);
+    const [selectedBranch, setSelectedBranch] = useState(origin.branch);
+    const [availableBranches, setAvailableBranches] = useState(currentOriginBranches);
+    const [branchLoadError, setBranchLoadError] = useState(currentBranchLoadError);
+    const [loadingBranches, setLoadingBranches] = useState(false);
+    const [error, setError] = useState(null);
+
+    const handeRemoteSelect = async (remote) => {
+        setSelectedRemote(remote);
+        setLoadingBranches(true);
+        remotes.listBranches(remote)
+                .then(newBranches => {
+                    setBranchLoadError(null);
+                    setAvailableBranches(newBranches);
+                    setLoadingBranches(false);
+
+                    if (newBranches.includes(origin.branch)) {
+                        setSelectedBranch(origin.branch);
+                    } else {
+                        setSelectedBranch(newBranches[0]);
+                    }
+                })
+                .catch(ex => {
+                    setBranchLoadError(ex.message);
+                    setAvailableBranches(null);
+                    setLoadingBranches(false);
+                });
+    };
+
+    const onRebaseClicked = () => {
+        onChangeRemoteOrigin(selectedRemote)
+                .then(() => onChangeBranch(selectedBranch))
+                .then(Dialogs.close())
+                .catch(ex => setError(ex.message));
+    };
+
+    const actions = [
+        <Button key="rebase"
+            variant="primary"
+            onClick={onRebaseClicked}
+        >
+            {_("Rebase")}
+        </Button>,
+        <Button key="cancel"
+            variant="link"
+            onClick={Dialogs.close}
+        >
+            {_("Cancel")}
+        </Button>
+    ];
+
+    const repositoryComponent = availableRemotes.length > 1
+        ? (
+            <Select isOpen={isSelectRemoteExpanded}
+                selections={selectedRemote}
+                onToggle={(_, expanded) => setSelectRemoteExpanded(expanded) }
+                onSelect={(_, remote) => { setSelectRemoteExpanded(false); handeRemoteSelect(remote) }}
+                menuAppendTo="parent"
+            >
+                {availableRemotes.map(remote => <SelectOption id={remote} value={remote} key={remote}> {remote} </SelectOption>)}
+            </Select>
+        )
+        : (
+            <Text>{availableRemotes[0]}</Text>
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-403-20231128-144720-91f0c34a-fedora-coreos-devel//Coverage/lcov/github-pr.diff.gcov.html)